### ISSUE TITLE
Cannot Log in - error "not enough values to unpack"

### ISSUE THREAD

**Issue by phreaq** [hassio-addons/addon-tautulli#155]
# Problem/Motivation

Unable to log into 2.3.0

## Expected behavior

ability to log in

## Actual behavior

after entering credentials, and selecting 'sign in', the login screen does nothing.

## Steps to reproduce

log out, log back in.

## Proposed changes

-

error logs:

2021-07-11 10:02:50 - ERROR :: CP Server Thread-12 : [11/Jul/2021:10:02:50] HTTP 
Traceback (most recent call last):
  File "/opt/lib/cherrypy/_cprequest.py", line 630, in respond
    self._do_respond(path_info)
  File "/opt/lib/cherrypy/_cprequest.py", line 689, in _do_respond
    response.body = self.handler()
  File "/opt/lib/cherrypy/lib/encoding.py", line 221, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/opt/lib/cherrypy/lib/jsontools.py", line 58, in json_handler
    value = cherrypy.serving.request._json_inner_handler(*args, **kwargs)
  File "/opt/lib/cherrypy/_cpdispatch.py", line 54, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/opt/plexpy/webauth.py", line 373, in signin
    valid_login, user_details, user_group = check_credentials(username=username,
  File "/opt/plexpy/webauth.py", line 135, in check_credentials
    if username == plexpy.CONFIG.HTTP_USERNAME and check_hash(password, plexpy.CONFIG.HTTP_PASSWORD):
  File "/opt/lib/hashing_passwords.py", line 51, in check_hash
    algorithm, hash_function, cost_factor, salt, hash_a = hash_.split('$')
ValueError: not enough values to unpack (expected 5, got 1)

**Comment by dreimer1986** [comment]
Comfirmed same here. Was able to temp. fix it with a 2.2.x backup.

**Comment by Nbr1Sniper** [comment]
Same here since day one… Can’t get into the web interface to set up fully with my Plex server…

**Comment by rpitera** [comment]
Having the same issue - similar error messages in log.  Strangely enough, the API pulls that I'm using are still working as REST sensors in HA and are showing current data.  Just can't log in to the UI.

**Comment by frenck** [comment]
I've confirmed the issue, however, I don't see the root cause yet.

**Comment by phreaq** [comment]
Is there a way to roll back to an older version? I used the Add-On Store to install Tautulli

**Comment by frenck** [comment]
> Is there a way to roll back to an older version? 

Using snapshots

**Comment by Nbr1Sniper** [comment]
I never had it before. This is my first time after finding a YouTube video review of the full application. I just want to have it run on my HA to limit overhead on my server.

**Comment by DigitalElise** [comment]
Also having the same issue as the OP.

**Comment by phixion** [comment]
It might be assumed, but if you set: 

1. set leave frontdoor open
2. launch the container
3. set a password in tautilli webinterface -> settings (in a seperate tab)
4. unset leave frontdoor open + set a username/password in HA supervisor and restart the container
5. refresh the tautilli webinterface -> settings tab

you can see that the username was handed over correctly but the password was left untouched

**Comment by Nbr1Sniper** [comment]
> It might be assumed, but if you manually set:
> 
> 1. set leave frontdoor open
> 2. launch the container
> 3. set a password in tautilli webinterface -> settings (in a seperate tab)
> 4. unset leave frontdoor open + set a username/password in HA supervisor and restart the container
> 5. refresh the tautilli webinterface -> settings tab
> 
> you can see that the username was handed over correctly but the password was left untouched

This doesn’t do anything for me. I could be doing it wrong but there isn’t a settings tab on the web interface as I can’t even log into it.

**Comment by phixion** [comment]
let me rephrase:
By adding

`leave_front_door_open: true`

to your tautulli configuration yaml (in homassistant > supervisor > tautulli > configuration) you will disable authentication for the webinterface, which then let's access tautulli without the need for a username and password. Aftwards you can access tautulli settings and witness the behaviour i was trying to explain.

**Comment by DigitalElise** [comment]
Doesn’t work for me, I still get the login screen and can’t get past that point even if I clear my browser or use a completely different device to access

**Comment by DigitalElise** [comment]
Alright, with a bit more rooting around I found that I needed to remove the username and password from the yaml or HomeAssistant supervisor settings, then restart. Once I did that it let me in and I could add the details back to the supervisor setup page.

**Comment by Nbr1Sniper** [comment]
Y’all are awesome! I got it to work… The trick for me was I had to install it, click leave open and then I could follow your steps. Before it wouldn’t let me change my username and pw blank.

**Comment by Nbr1Sniper** [comment]
I investigated more; I have to sign in using Plex… I can’t local log in

**Comment by rpitera** [comment]
> Alright, with a bit more rooting around I found that I needed to remove the username and password from the yaml or HomeAssistant supervisor settings, then restart. Once I did that it let me in and I could add the details back to the supervisor setup page.

@DigitalElise - Can you explain where you did this?  Because the configuration in the add-on doesn't allow you to save without the username/password lines or by leaving them blank, unless I'm missing something obvious (which is entirely possible, lol).

**Comment by rpitera** [comment]
>Uninstall and reinstall the add on. That’s what I did.

Don't you lose all the viewing data then?  I'd hate to do that and then find out a couple of days later that @frenck had found a solution.  I think I'll wait a while longer and then if it doesn't seem like a fix is going to happen, I'll try that as a work-around.

Thanks for tip!

**Comment by Nbr1Sniper** [comment]
Try using leave the front door open in the hidden options. Delete the log in data. Then log into the site and after that’s all set up go back in and turn the front door feature off. 

It didn’t bother me to delete it and add it as it was the first time using it.

**Comment by Nbr1Sniper** [comment]
> > Uninstall and reinstall the add on. That’s what I did.
> 
> Don't you lose all the viewing data then? I'd hate to do that and then find out a couple of days later that @frenck had found a solution. I think I'll wait a while longer and then if it doesn't seem like a fix is going to happen, I'll try that as a work-around.
> 
> Thanks for tip!

Try using leave the front door open in the hidden options. Delete the log in data. Then log into the site and after that’s all set up go back in and turn the front door feature off. 

It didn’t bother me to delete it and add it as it was the first time using it.

**Comment by edalquist** [comment]
For other YAML novices I had to set username and password to empty strings to get this work-around working.

```
username: ''
password: ''
leave_front_door_open: true
```

**Comment by rpitera** [comment]
Thanks, @edalquist!  That'll work for now until @frenck figures out what's failing.  And I didn't have to lose anything in the process.

**Comment by Nbr1Sniper** [comment]
My other issue I have is I have to use plex to log into the system. Local login won’t work. I can’t get the iframe/web front end to work as it won’t redirect you to plex to log in. I don’t want to keep the front door open for security. 

Thoughts?

**Comment by rpitera** [comment]
Still an issue.  Workaround is to leave the front door open or reinstall and then leave front door open - neither of which are desirable.  Please leave open!

**Comment by SiliconAvatar** [comment]
This issue is not not resolved. There have been no updates to this addon in the 2 months since this issue was reported, so not fixed in a newer version.
This issue is not stale.

**Comment by dreimer1986** [comment]
Still the same here. I tend to close the front door afterwards, but this only helps as long as the session did not timeout.

**Comment by ankorez** [comment]
> let me rephrase: By adding
> 
> `leave_front_door_open: true`
> 
> to your tautulli configuration yaml (in homeassistant > supervisor > tautulli > configuration) you will disable authentication for the webinterface, which then let's access tautulli without the need for a username and password.
> 
> Afterwards you can access tautulli settings and witness the behavior i was trying to explain. In fact re-enabling authentication in HA while leaving the tautulli tab open in your browser will leave you logged on and you can see the username is being updated but not the password.

Hey ! thanks for the tips !

**Comment by datbilling** [comment]
I have the same issue. Any news on a fix?

**Comment by dreimer1986** [comment]
Right now the "fix" is opening the front door, do your things and then close it again. With reboots after each step of course. Ugly but at least secure if the instance is externally available.

**Comment by tarheelz** [comment]
Thanks all here for their work.  Got it working using the advice in this thread.

**Comment by flecmart** [comment]
Still an issue.

github-actions[bot] ***@***.***> schrieb am Mo., 27. Dez.
2021, 09:22:

**Comment by reubadoob** [comment]
`2022-01-14 16:05:44 - ERROR :: CP Server Thread-7 : [14/Jan/2022:16:05:44] HTTP 
Traceback (most recent call last):
  File "/opt/lib/cherrypy/_cprequest.py", line 630, in respond
    self._do_respond(path_info)
  File "/opt/lib/cherrypy/_cprequest.py", line 689, in _do_respond
    response.body = self.handler()
  File "/opt/lib/cherrypy/lib/encoding.py", line 221, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/opt/lib/cherrypy/lib/jsontools.py", line 58, in json_handler
    value = cherrypy.serving.request._json_inner_handler(*args, **kwargs)
  File "/opt/lib/cherrypy/_cpdispatch.py", line 54, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/opt/plexpy/webauth.py", line 373, in signin
    valid_login, user_details, user_group = check_credentials(username=username,
  File "/opt/plexpy/webauth.py", line 135, in check_credentials
    if username == plexpy.CONFIG.HTTP_USERNAME and check_hash(password, plexpy.CONFIG.HTTP_PASSWORD):
  File "/opt/lib/hashing_passwords.py", line 51, in check_hash
    algorithm, hash_function, cost_factor, salt, hash_a = hash_.split('$')
ValueError: not enough values to unpack (expected 5, got 1)`

Same errors as others have mentioned above. PW word matches config fails to login. Generates error above.

**Comment by dlm21** [comment]
I ran into this today setting up a fresh hass.io install. Was able to get around it with the fixes already mentioned, but reproducing this is easy at least, so hopefully a fix can show up at some point

**Comment by ccnpsmart** [comment]
Does anyone know if there is a fix coming for this?

**Comment by stu1811** [comment]
> It might be assumed, but if you manually set:
> 
> 1. set leave frontdoor open
> 2. launch the container
> 3. set a password in tautilli webinterface -> settings (in a seperate tab)
> 4. unset leave frontdoor open + set a username/password in HA supervisor and restart the container
> 5. refresh the tautilli webinterface -> settings tab
> 
> you can see that the username was handed over correctly but the password was left untouched

This workaround is still valid. Thanks for posting.

**Comment by SiliconAvatar** [comment]
Issue is not stale and not resolved.

**Comment by dreimer1986** [comment]
Confirmed not fixed at all.

**Comment by Hostground** [comment]
Also having the same problem.

**Comment by SiliconAvatar** [comment]
Issue is not stale and not resolved.

**Comment by SiliconAvatar** [comment]
Issue is still not resolved.
